sfaira/data/databases/cellxgene_loader.py

```diff
@@ -0,0 +1,75 @@
+import numpy as np
+import os
+from typing import Union
+from .external import DatasetBase
+import anndata
+
+from .external import DatasetGroupBase
+
+
+class Dataset(DatasetBase):
+    """
+    This is a dataloader for downloaded h5ad from cellxgene.
+
+    :param path:
+    :param meta_path:
+    :param kwargs:
+    """
+
+    def __init__(
+            self,
+            path: Union[str, None],
+            fn: str,
+            meta_path: Union[str, None] = None,
+            **kwargs
+    ):
+        DatasetBase.__init__(self=self, path=path, meta_path=meta_path, **kwargs)
+        self.fn = fn
+        self.species = str(fn).split("_")[2]
+        self.id = str(fn).split(".")[0]
+        self.organ = str(fn).split("_")[3]
+        self.sub_tissue = None
+        self.download_website = None  # TODO
+        self.has_celltypes = True
+
+        self.class_maps = {
+            "0": {},
+        }
+
+    def _load(self, fn=None):
+        fn = os.path.join(self.path, self.fn)
+        adata = anndata.read(fn)
+        adata.X = adata.raw.X
+
+        self.adata.uns["lab"] = adata.uns["contributors"]["name"]
```

Comment:
Looks like this loader evaded the refactor.

Reply:
thanks @ambrosejcarr, indeed this was the case, I fixed that and introduced a separate constants container for cellxgene objects!

Reply:
I like this implementation. I made my comment because I assumed (incorrectly) that you would _replace_ our schema names with those in `ADATA_IDS`, making our data conform to your internal schema. 

Checking my understanding: What you've done instead is added cellxgene's schema as an alternative schema. Is that correcct? 

If going this route, I'd recommend you inherit both `ADATA_IDS` and `ADATA_IDS_CELLXGENE` from an abstract base class that defines the properties that `sfaira` *requires*. I imagine this includes but may not be limited to the "lazy loading metadata". If you did it this way, a user could choose to swap between different schema by importing alternative data loader classes. Someone who wanted to use `sfaira` to work in the HTAN or HuBMAP schema would simply need to write a data loader for their ecosystem...

Reply:
> Checking my understanding: What you've done instead is added cellxgene's schema as an alternative schema. Is that correct?

yes! I also refactored the classes to be derived of base classes that reflect what we require for other datat bases now! this closesly ties in with the meta data as well, correct!

Reply:
Extended this - all relevant properties for lazy loading are now exactly those that are also read from meta data files. So in this case, we just need the matched meta data files which are then queried during lazy loading. These match a subset of all generally required properties of the object, essentially the most important dataset-wise (as oppose to cell-wise) properties. If it s the case that a data set has cells from different organs for example, the organ entry of this meta data file would just be a list of contained organs, but not cell-wise.

Reply:
On 2nd thoughts, I dont think it s really necessary that a 3rd party also maintains the meta files - if the adata objects are streamlined as this is outlined here now, these relevant attributes can be easily read from a loaded adata object and an up-to-date library of meta files can be built locally with write_meta() once before lazy loading usage. This is not very costly (this is how we maintain our meta data file library for example) and reduces overhead of interfacing 3rd parties a lot, let me know if somebody disagrees.

Reply:
This looks great to me. I don't understand what this means, however: 

> Extended this - all relevant properties for lazy loading are now exactly those that are also read from meta data files. So in this case, we just need the matched meta data files which are then queried during lazy loading

Reply:
Previously, meta data files were not directly linked to lazy loading as we largely set properties for lazy loading in datasest constructors. Initially we only used meta data files to generate data base statistics actually. I streamlined this now so that we dont have different versions of meta data files.